bin/test-workflow/8_app_verify_authentication.sh

```diff
@@ -72,38 +73,61 @@ pod_curl() {
 echo "Deploying a test curl pod"
 deploy_test_curl
 echo "Waiting for test curl pod to become available"
-bl_retry_constant "${RETRIES}" "${RETRY_WAIT}"  check_test_curl
-  
+bl_retry_constant "${RETRIES}" "${RETRY_WAIT}" check_test_curl
+
 echo "Waiting for pods to become available"
 
+# restore array of apps to run
+declare -a install_apps=($(split_on_comma_delimiter $INSTALL_APPS))
+
 if [[ "$PLATFORM" == "openshift" ]]; then
-  sidecar_pod=$(get_pod_name test-app-summon-sidecar)
-  secretless_pod=$(get_pod_name test-app-secretless)
-  secrets_provider_standalone_pod=$(get_pod_name test-app-secrets-provider-standalone)
-  secrets_provider_init_pod=$(get_pod_name test-app-secrets-provider-init)
-
-  # Routes are defined, but we need to do port-mapping to access them
-  oc port-forward "$sidecar_pod" 8081:8080 > /dev/null 2>&1 &
-  SIDECAR_PORT_FORWARD_PID=$!
-  oc port-forward "$secretless_pod" 8083:8080 > /dev/null 2>&1 &
-  SECRETLESS_PORT_FORWARD_PID=$!
-  oc port-forward "$secrets_provider_standalone_pod" 8084:8080 > /dev/null 2>&1 &
-  SECRETS_PROVIDER_STANDALONE_PID=$!
-  oc port-forward "$secrets_provider_init_pod" 8086:8080 > /dev/null 2>&1 &
-  SECRETS_PROVIDER_INIT_PORT_FORWARD_PID=$!
+  # Routes are defined, but we need to do port-mapping to access them.
+  # Port-mapping needs to be conditional - the process kill loop in function 'finish'
```

Comment:
Ouchy!!! Nice find!

Since the problem has been nicely fixed, can we delete the explanation "the process kill loop in function 'finish' causes nearly invisible failures tying to kill a process that isn't deployed."?

Seems to me that doing the port-mapping conditionally is just the right thing to do, and should have been conditional from day 1.